tests/e2e/test_fetcher/attribute_values.py

```diff
@@ -0,0 +1,174 @@
+#
+# Copyright (c) 2025, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import datetime
+import functools as ft
+from collections.abc import Iterable
+from typing import (
+    Any,
+    Optional,
+)
+
+from neptune_api.client import AuthenticatedClient
+from neptune_retrieval_api.api.default import query_attributes_within_project_proto
+from neptune_retrieval_api.models import QueryAttributesBodyDTO
+from neptune_retrieval_api.proto.neptune_pb.api.v1.model.attributes_pb2 import ProtoQueryAttributesResultDTO
+from neptune_retrieval_api.proto.neptune_pb.api.v1.model.leaderboard_entries_pb2 import ProtoAttributeDTO
+
+from . import (
+    identifiers,
+    paging,
+)
+
+
+def fetch_attribute_values(
+    client: AuthenticatedClient,
+    project: identifiers.ProjectIdentifier,
+    *,
+    attributes: Iterable[identifiers.AttributePath],
+    custom_run_id: Optional[identifiers.CustomRunId] = None,
+    run_id: Optional[identifiers.SysId] = None,
+) -> dict[identifiers.AttributePath, Any]:
+    attribute_set: set[identifiers.AttributePath] = set(attributes)
+
+    if not attribute_set:
+        return {}
+
+    if custom_run_id is not None:
+        experiment_id = f"CUSTOM/{project}/{custom_run_id}"
```

Comment:
I'd prefer not to use `experiment_*` as it's an entity in our domain. Even if the API parameter is named so unfortunately.